recipes/nncf/meta.yaml

```diff
@@ -0,0 +1,57 @@
+{% set name = "nncf" %}
+{% set version = "2.0.0" %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 8bdfabaf1b163da6e80dc2d7906c2c8342567cff28c40e97c2cb65929a750736
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - addict >=2.4.0
+    - jsonschema ==3.2.0
+    - jstyleson >=0.0.2
+    - matplotlib-base >=3.3.4  # [py>=37]
+    - natsort >=7.1.0
+    - networkx >=2.5
+    - ninja >=1.10.0.post2
+    - numpy ~=1.19.2
+    - pandas >=1.1.5  # [py>=37]
+    - pydot >=1.4.1
+    - python
+    - scikit-learn >=0.24.0
+    - scipy <=1.5.4,>=1.3.2  # [py<37]
+    - texttable >=1.6.3
+    - pytorch >=1.5.0,<=1.8.1,!=1.8.0
+    - tqdm >=4.54.1
+    - wheel >=0.36.1
+    - {{ compiler('c') }}
```

Comment:
maybe better to depend on the `compilers` or `cxx-compiler` / `c-compiler` package?

Reply:
I am not sure how to do that. Could you provide code suggestions? Thank you!

Reply:
you can just use the `compilers` package (or more accurately the `c-compiler`, `cxx-compiler`. But maybe it's not necessary for this case.

Reply:
why do you need the compilers actually? I didn't see any reference to e.g. GCC in the nncf source code

Reply:
@wolfv this is because NNCF utilizes the underlying PyTorch's JIT compilation of custom kernels which we include in the NNCF package as source files (both .cpp and .cu for the CPU and GPU execution cases). Therefore it is rather torch and the way it is used by NNCF that enforces the necessity of a C++ compiler on the target machine

Reply:
I agree with @wolfv here that you should depend on c-compiler and cxx-compiler here.

Reply:
I disagree with @wolfv and @beckermr . `c-compiler` and `cxx-compiler` should only be a user thing. It should not be used as a requirement for a package.

Reply:
@beckermr, this was not resolved

Reply:
Ohhhh sorry! I thought it was and merged it. We can fix this up in the feedstock. My apologies!